content/spin/v3/community-kinesis-trigger.md

```diff
@@ -0,0 +1,131 @@
+
+<!-- @nocpy -->
+
+```toml
+[[trigger.kinesis]]
+stream_arn = "arn:aws:kinesis:us-east-1:1234567890:stream/TestStream"
+batch_size = 1000
+shard_idle_wait_millis = 250
+detector_poll_millis = 30000
+shard_iterator_type = "TRIM_HORIZON"
+component = "hello"
+```
+
+> Note: See [the list of all available configuration variables](https://github.com/ogghead/spin-trigger-kinesis?tab=readme-ov-file#spintoml) for the AWS Kinesis trigger.
+
+## Installing the AWS Kinesis Trigger
+
+### Installing the AWS Kinesis Trigger Plugin
+
+First, we install the plugin:
+
+```bash
+spin plugins update
+spin plugins install trigger-kinesis
+```
+
+### Installing the AWS Kinesis Trigger Template
+
+Then, we install the template:
+
+```bash
+spin templates install --git https://github.com/ogghead/spin-trigger-kinesis
+```
+
+## Creating the Application
+
+With the plugin and template installed, we create a new application:
+
+```bash
+spin new -t kinesis-rust hello --accept-defaults
+```
+
+As part of creating the new application, `spin` CLI will ask for the Amazon Resource Name (ARN). When asked, provide the ARN of your AWS Kinesis instance and confirm it with `[ENTER]`. 
+
+### Inspecting the Source Code
+
+The Rust source code for this application is as follows:
+
+```Rust
+use spin_kinesis_sdk::{kinesis_component, Error, KinesisRecord};
+
+#[kinesis_component]
+async fn handle_batch_records(records: Vec<KinesisRecord>) -> Result<(), Error> {
+    for record in records {
+        println!("I GOT A RECORD!  ID: {:?}", record.sequence_number);
+        let data = String::from_utf8(record.data.inner).unwrap();
+        println!("  ... DATA: {:?}", data);
+    }
+
+    Ok(())
+}
+
+```
+
+## Building and Running the Application
+
+We can immediately run this pre-written (template) application and observe the time-driven execution:
+
+```bash
+cd hello
+spin build --up
+
+Building component hello with `cargo build --target wasm32-wasi --release`
+
+...
+
+Finished building all Spin components
+Logging component stdio to ".spin/logs/"
+I GOT A RECORD!  ID: 101
+  ... DATA: { "value" : "foo" }
+I GOT A RECORD!  ID: 102
+  ... DATA: { "value" : "bar" }
+```
+
+As we can see from the above output, our application is now running and executing the function for each data record sent through your individual AWS Kinesis instance.
+
+If you would like to learn more about using the Spin Kinesis Trigger, please check out the ["How I wrote A Kinesis Trigger PLugin For Spin"](https://www.fermyon.com/blog/how-i-wrote-a-kinesis-trigger-plugin-for-spin) blog post and consult the [Spin Kinesis Trigger GitHub repository](https://github.com/ogghead/spin-trigger-kinesis/).
```

Comment:
```suggestion
If you would like to learn more about using the Spin Kinesis trigger, please check out the ["How I wrote A Kinesis Trigger Plugin For Spin"](https://www.fermyon.com/blog/how-i-wrote-a-kinesis-trigger-plugin-for-spin) blog post and consult the [Kinesis trigger GitHub repository](https://github.com/ogghead/spin-trigger-kinesis/).
```

(this is mostly tiny typos but I did realise I was getting weary of reading the phrase "Spin Whatever trigger".  We're already in the context of Spin - do we need to mention it every time?  Or is this some SEO thing?)